mantidproject/mantid
# Crop phasequad by first and last good data

**Description of work.**

This adds a cropping step before the creation of a PhaseQuad workspace in order to remove erroneous peaks see issue #25172 for further details.

This also adds a masking step before both the creation of the phase quad workspace and the calculation of the phase table to mask out detectors with erroneous values which can throw off the fit.

The calculation of PhaseQuad is updated to that it ignores spectrum for which the PhaseQuad fit has failed.

CalMuonDetector phases is also updated so that when the fit on a single spectrum fails a warning is output rather than the whole fit being cancelled.

**Report to:** James Lord 

**To test:**
* You will unfortunately need access to the ISIS data archive.
* Open frequency domain analysis, select Chronus as your instrument and load run 7873. 
* Go to the transform tab choose FFT change last good data to 10 and click calculate
* Plot the resulting workspace and check it does not have a peak at negative frequency



Fixes #25172  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
